client/dashboard/app/queries/site-staging-sites.ts

```diff
@@ -1,9 +1,26 @@
-import { createStagingSite, deleteStagingSite } from '../../data/site-staging-site';
+import {
+	createStagingSite,
+	deleteStagingSite,
+	fetchStagingSiteData,
+} from '../../data/site-staging-site';
+
+export const STAGING_SITE_DELETE_MUTATION_KEY = 'staging-site-delete-mutation';
 
 export const stagingSiteCreateMutation = ( siteId: number ) => ( {
 	mutationFn: () => createStagingSite( siteId ),
 } );
 
 export const stagingSiteDeleteMutation = ( stagingSiteId: number, productionSiteId: number ) => ( {
 	mutationFn: () => deleteStagingSite( stagingSiteId, productionSiteId ),
+	mutationKey: [ STAGING_SITE_DELETE_MUTATION_KEY, stagingSiteId ],
+} );
+
+export const stagingSiteDeleteStatusQuery = ( productionSiteId: number ) => ( {
+	queryKey: [ 'staging-site-delete-status', productionSiteId ],
+	queryFn: () => fetchStagingSiteData( productionSiteId ),
+	refetchInterval: 3000,
```

Comment:
Can we put this in the component instead, not here? This behavior is specific to the banner.